src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor({ health = 100, name }) {
+    this.health = health;
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(bittedAnimal) {
+    if (bittedAnimal instanceof Herbivore && !bittedAnimal.hidden) {
+      bittedAnimal.health -= 50;
+    }
+
+    if (bittedAnimal.health <= 0) {
+      Animal.alive = Animal.alive.filter((animal) => animal !== bittedAnimal);
+    }
```

Comment:
The check for the animal's health should be inside the conditional block that checks if the bitten animal is a Herbivore and not hidden. This will prevent unnecessary checks on Carnivores or when the Herbivore is hidden.